`init-declarations` false positive at `for-in/of`.

## test.js

```js
/* eslint init-declarations: 2 */
var a = 0;

for (var b in []) {}
```

## Command & Actual

```
> eslint -v
v1.0.0-rc-3
> eslint test.js --no-eslintrc

test.js
  4:10  error  Variable 'b' should be initialized on declaration  init-declarations

✖ 1 problem (1 error, 0 warnings)
```

## Expected

No errors.

--- a/lib/rules/init-declarations.js
+++ b/lib/rules/init-declarations.js
@@ -7,6 +7,25 @@
 "use strict";
 
 //------------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------------
+
+/**
+ * Checks whether or not a given declarator node has its initializer.
+ * @param {ASTNode} node - A declarator node to check.
+ * @returns {boolean} `true` when the node has its initializer.
+ */
+function isInitialized(node) {
+    var declaration = node.parent;
+    var block = declaration.parent;
+
+    if (block.type === "ForInStatement" || block.type === "ForOfStatement") {
+        return block.left === declaration;
+    }
+    return node.init != null;
+}
+
+//------------------------------------------------------------------------------
 // Rule Definition
 //------------------------------------------------------------------------------
 
@@ -22,7 +41,7 @@ module.exports = function(context) {
     //--------------------------------------------------------------------------
 
     return {
-        "VariableDeclaration": function(node) {
+        "VariableDeclaration:exit": function(node) {
 
             var kind = node.kind,
                 declarations = node.declarations;
@@ -30,7 +49,7 @@ module.exports = function(context) {
             for (var i = 0; i < declarations.length; ++i) {
                 var declaration = declarations[i],
                     id = declaration.id,
-                    initialized = declaration.init !== null;
+                    initialized = isInitialized(declaration);
 
                 if (id.type !== "Identifier") {
                     continue;

--- a/tests/lib/rules/init-declarations.js
+++ b/tests/lib/rules/init-declarations.js
@@ -22,6 +22,8 @@ eslintTester.addRuleTest("lib/rules/init-declarations", {
         "function foo() { var foo = 0; var bar = []; }",
         "var fn = function() {};",
         "var foo = bar = 2;",
+        "for (var foo in []) {}",
+        {code: "for (var foo of []) {}", ecmaFeatures: {forOf: true}},
         {
             code: "let a = true;",
             ecmaFeatures: {
@@ -117,9 +119,16 @@ eslintTester.addRuleTest("lib/rules/init-declarations", {
     invalid: [
         {
             code: "var foo;",
-            ecmaFeatures: {
-                blockBindings: true
-            },
+            options: ["always"],
+            errors: [
+                {
+                    message: "Variable 'foo' should be initialized on declaration.",
+                    type: "VariableDeclarator"
+                }
+            ]
+        },
+        {
+            code: "for (var a in []) var foo;",
             options: ["always"],
             errors: [
                 {

